### PR TITLE
Add helper to find preferredData by regex and add to Analyzer

### PR DESCRIPTION
**Pull request by fbruton**

### CODE REVIEW

File: src/main/java/emissary/core/IBaseDataObjectHelper.java [hunk 1]
```diff
@@ -215,4 +218,40 @@ public static void addParentInformationToChildren(final IBaseDataObject parent,
             }
         }
     }
+
+    /**
+     * Search for the first preferred view by regular expression or use the primary data if none match
+     *
+     * @param payload the payload to pull data from
+     */
+    public static byte[] findPreferredDataByRegex(final IBaseDataObject payload, List<Pattern> preferredViewNamePatterns) {
```
Comment (drivenflywheel) [inline src/main/java/emissary/core/IBaseDataObjectHelper.java:227]: Why prefix this method name with "find" but prefix the non-regex one with "get"?  Aside from the pattern usage, they're provide very parallel functionality, with both falling back to using `data()` is a suitable view isn't present.

File: src/test/java/emissary/core/IBaseDataObjectHelperTest.java [hunk 2]
```diff
@@ -353,4 +356,39 @@ void testAddParentInformationToChildren() {
 
         assertTrue(true);
     }
+
+    @Test
+    void testGetPreferredData() {
```
Comment (drivenflywheel) [inline src/test/java/emissary/core/IBaseDataObjectHelperTest.java:361]: Test name (`testGetPreferredData`) doesn't match the method-under-test (`findPreferredDataByRegex`)

File: src/main/java/emissary/analyze/Analyzer.java [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
 
 /**
  * Base class for analyzers
  */
 public abstract class Analyzer extends ServiceProviderPlace {
-    protected List<String> PREFERRED_VIEWS;
-    protected String ANALYZED_DATA_NAME;
+    protected List<String> preferredViews;
+    protected String analyzedDataName;
+    protected List<Pattern> perferredViewPatternList;
```
Comment (drivenflywheel) [inline src/main/java/emissary/analyze/Analyzer.java:19]: Minor typo.  Might also consider dropping the `List` suffix and just pluralizing to `preferredViewPatterns`. 
```suggestion
    protected List<Pattern> preferredViewPatternList;
```

**Review by drivenflywheel — COMMENTED** [review]
Minor suggestions, only cosmetic.

File: src/main/java/emissary/core/IBaseDataObjectHelper.java (hunk 1 above)
Comment (fbruton) [reply src/main/java/emissary/core/IBaseDataObjectHelper.java:227]: The "get" method name was a direct copy from the `Analyze` class. I'll update the method name to be consistent.

**Review by drivenflywheel — APPROVED** [review]
LGTM  :rocket:

**Review by dev-mlb — APPROVED** [review]

**Pull request merged**